test/discord.test.ts

```diff
@@ -293,161 +294,407 @@ describe('calling getUserId', () => {
+            expect.assertions(2);
+                
+            const client = setupClient();
+
+            const projectId = 0;
+            const error = 'error';
+            const channelName = 'channel';
+
+            let sentMessageValue = '';
+
+            const loggerMock = new LoggerMockBuilder()
+                .withWarn(jest.fn().mockImplementation(value => {
+                    expect(value).toBe(`Error creating task: ${error}`);
+                    expect(sentMessageValue).toBe(`Unable to find ActiveCollab project for channel ` 
+                        + channelName);
+                    done();
+                }))
+                .build();
+
+            const mappingControllerMock = new MappingControllerMockBuilder()
+                .withGetProjectId(jest.fn(() => { throw Error(error); }))
+                .build();
+
+            const discordController = new DiscordControllerBuilder()
+                .withMappingController(mappingControllerMock)
+                .withClient(client)
+                .withLogger(loggerMock)
+                .build();
+
+            const message = new MessageBuilder()
+                .withContent('!tasks create task')
+                .withSend(jest.fn(async value => {
+                    sentMessageValue = value;
+                }))
+                .withChannelName(channelName)
+                .build();
+
+            client.emit('message', message);
+        });
     });
-
-    it('should call commandController.tasksDueThisWeekForProject when command is' 
-        + ' "!tasks due" and command sent from project channel', done => {
-        const client = setupClient();
-
-        const commandControllerMock = new CommandControllerMockBuilder()
-            .withTasksDueThisWeekForProject(jest.fn(async () => done()))
-            .build();
-
-        const mappingControllerMock = new MappingControllerMockBuilder()
-            .withGetProjectId(jest.fn().mockReturnValue(0))
-            .build();
-
-        const discordController = new DiscordControllerBuilder()
-            .withMappingController(mappingControllerMock)
-            .withCommandController(commandControllerMock)
-            .withClient(client)
-            .build();
-
-        const sentMessage = {
-            edit: jest.fn(() => Promise.resolve())
-        };
-
-        const message = {
-            content: '!tasks due',
-            author: 'author',
-            channel: {
-                send: jest.fn(async () => sentMessage),
-                startTyping: jest.fn(() => Promise.resolve()),
-                stopTyping: jest.fn(() => Promise.resolve()),
-                name: 'channel',
-                type: 'text'
-            }
-        };
-
-        client.emit('message', message);
+    
+    describe('when message is "!tasks due"', () => {   
+        it('should call commandController.tasksDueThisWeekForProject when command is' 
+            + ' sent from project channel', done => {
```

Comment:
This test has no assertions, is that correct? Where possible, I think checking that the jest stub has been called makes the intent of the test more obvious than calling a stub with a callback function that invokes `done()`.